server/weaverbird/backends/sql_translator/steps/uniquegroups.py

```diff
@@ -0,0 +1,66 @@
+from distutils import log
+
+from weaverbird.backends.sql_translator.steps.utils.query_transformation import (
+    build_selection_query,
+    complete_fields,
+)
+from weaverbird.backends.sql_translator.types import (
+    SQLPipelineTranslator,
+    SQLQuery,
+    SQLQueryDescriber,
+    SQLQueryRetriever,
+)
+from weaverbird.pipeline.steps import UniqueGroupsStep
+
+
+def translate_uniquegroups(
+    step: UniqueGroupsStep,
+    query: SQLQuery,
+    index: int,
+    sql_query_retriever: SQLQueryRetriever = None,
+    sql_query_describer: SQLQueryDescriber = None,
+    sql_translate_pipeline: SQLPipelineTranslator = None,
+) -> SQLQuery:
+    query_name = f'UNIQUEGROUPS_STEP_{index}'
+
+    log.debug(
+        "############################################################"
+        f"query_name: {query_name}\n"
+        "------------------------------------------------------------"
+        f"step.name: {step.name}\n"
+        f"step.on: {step.on}\n"
+        f"query.transformed_query: {query.transformed_query}\n"
+        f"query.metadata_manager.tables_metadata: {query.metadata_manager.tables_metadata}\n"
+        f"query.metadata_manager.query_metadata: {query.metadata_manager.query_metadata}\n"
+    )
+
+    select_query = complete_fields(query=query)
+    group_by_query: str = ""
+    if len(step.on) > 0:
+        # We build the group by query part
+        for index, gb in enumerate(step.on):
+            group_by_query += (" GROUP BY " if index == 0 else ", ") + gb
```

Comment:
Ta syntaxe est ok. 
Je pense qu'on peut aussi utiliser:
`group_by_query = f" GROUP BY {', '.join([gb for gb in step.on])}  "`
A la place des lignes 41 & 42